client/landing/gutenboarding/components/header/index.tsx

```diff
@@ -81,25 +85,26 @@ const Header: FunctionComponent< Props > = ( { prev } ) => {
 		</span>
 	);
 
-	const siteUrl = currentDomain?.domain_name || siteTitle || currentUser?.username;
-	const siteCreationData = {
-		siteTitle,
-		siteVertical,
-		...( siteUrl && { siteUrl } ),
-		...( selectedDesign?.slug && { theme: selectedDesign?.slug } ),
-	};
-
-	const handleCreateSite = () => {
-		setIsCreatingSite( true );
-		history.push( Step.CreateSite );
-		createSite( siteCreationData ).then( siteSlug => {
-			resetOnboardStore();
-			window.location.href = `/block-editor/page/${ siteSlug }/home?is-gutenboarding`;
+	const handleCreateSite = ( username: string, bearerToken?: string ) => {
+		const siteUrl = currentDomain?.domain_name || siteTitle || username;
+		createSite( {
+			blog_name: siteUrl?.split( '.wordpress' )[ 0 ],
+			blog_title: siteTitle,
+			options: {
+				site_vertical: siteVertical?.id,
+				site_vertical_name: siteVertical?.label,
+				site_information: {
+					title: siteTitle,
+				},
+				site_creation_flow: 'gutenboarding',
+				theme: `pub/${ selectedDesign?.slug }`,
+			},
+			...( bearerToken && { authToken: bearerToken } ),
 		} );
 	};
 
-	if ( shouldCreate && currentUser ) {
-		handleCreateSite();
+	if ( shouldCreate && newUser && newUser.bearerToken && newUser.username ) {
+		handleCreateSite( newUser.username, newUser.bearerToken );
 		setShouldCreate( false );
 	}
```

Comment:
I am trying to work out if this should be wrapped in a `useEffect( () => {     ....      }, [shouldCreate])` block, the docs say you should use `useEffect` when generating side effects, e.g. making http calls inside a FunctionComponent, which `createSite` does.  [useEffect docs](https://reactjs.org/docs/hooks-effect.html). we did use `useEffect` to wrap the `setDomain` call on line `63`.

I'm not sure what the implication of not using `useEffect` would be however? @razvanpapadopol do you know more?

Reply:
This looks like exactly the type of thing that should go into `useEffect`, good spot 👍 

> I'm not sure what the implication of not using `useEffect` would be however? @razvanpapadopol, @sirreal do you know more?

`useEffect` is going to ensure that the function it is passed is only going to be invoked when the dependencies change. Because function components are effectively a single `render` function, the code in them should be safe to run repeatedly. Side effects —like creating a new site— should often only be run under certain conditions and not any number of times, `useEffect` can help here.